### DISCUSSION TITLE
Moving from G6 to G7 on S24 Ultra and xDrip+ is failing to understand data (Firmware version issue?)

### DISCUSSION THREAD

**Discussion by hax90** [NightscoutFoundation/xDrip#3489] [Q&A] [answered]
I have been using the g6 for a while but I decided to try out the g7, hoping the xDrip kinks would be worked out by now.

I followed the guide here: https://navid200.github.io/xDrip/docs/Dexcom/G7.html

I got the pairing request and the 30 minute countdown but when the countdown ended, the system status page something about unrecognized FW version and I needed to update. I clicked it but decided not to install (it was trying to install nightly) without looking at the status first. When I went back to status, I see "32.192.106.0 Unknown! Tap to report". I went ahead and installed the nightly release and that's still all I see. When I go to Events Log, I just get "Ob1G5StateMachine: New BgReading was null in processGlucoseRxMessage!" every 5 minutes. The home page still shows "Verify settings and wait for connectivity".
![Screenshot_20240527_002627_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/170917084/3935a119-d470-4065-bbaf-0089cc304955)
![Screenshot_20240527_002645_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/170917084/1bc7a35a-7c6d-42d7-ac7b-f02fde6f5b98)

**Comment by hax90** [comment]
Here are my settings to just confirm I followed the guide correctly:
![Screenshot_20240527_003144_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/170917084/e452738d-30c6-4df6-92e8-3c83a7b5abbe)
![Screenshot_20240527_003152_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/170917084/3c49c7cd-8e82-411e-8b36-4e52f253b73c)
![Screenshot_20240527_003436_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/170917084/94ea84b7-19af-4113-9785-e68083a37313)
![Screenshot_20240527_003511_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/170917084/bd1214cd-0fe2-4675-a587-d52d1b70337a)
![Screenshot_20240527_003513_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/170917084/f670cd35-e82a-49f2-8707-17d9c995a814)
![Screenshot_20240527_003533_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/170917084/9f146ee3-8e87-4370-ad9d-04930f2233f1)

**Comment by hax90** [comment]
To be clear, I just installed the Dexcom G7 app for the first time and I'm seeing readings there without issue.

**Comment by hax90** [comment]
Here is the latest in my debug play-by-play:

After opening xdrip back up after installing and seeing readings on the Dexcom app, I saw a popup that said autostarting sensor or something. The status page has changes to look like this:

![Screenshot_20240527_011017_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/170917084/df7ba67f-2abe-492d-9f5e-a8f4c2ebea57)

**Comment by hax90** [comment]
And out of nowhere, it has begun showing data... ok now I'm just confused. Can anyone give insight into what happened? I'm concerned this will happen again in the future.

**Reply by hax90** [reply]
Andddddd it's dead again. Same errors as my most recent screenshot and no readings for 16 minutes. I'm assuming the Dexcom app and xDrip aren't playing well together? I only installed the Dexcom app after the issues began though so I don't really know the real root cause.

**Reply by Navid200** [reply]
You can use either the Dexcom app or xDrip as the collector, but not both.

**Comment by Navid200** [comment, marked as answer]
This is an excellent presentation of the problem.  Thanks for that.

We have a bug in the latest Nightly.  I have an open PR for it: https://github.com/NightscoutFoundation/xDrip/pull/3476
While we wait for this to be resolved, anyone coming from a G6 can either wait 2 hours for their readings to start only for their very first G7.
Or, they can temporarily go back to the previous Nightly so that they can start the session and then, come back to the latest Nightly.

This problem will never happen again when you get to the point you will be switching from a G7 to another G7 even while we are waiting for the PR to be merged.

**Reply by Navid200** [reply]
If you decide to use xDrip, and if it unpairs on its own, it is because of Samsung and you should enable `Special pairing workaround`.

**Reply by hax90** [reply]
So you're saying that if I uninstall Dexcom's app, I should be good to go (even with the current sensor)? I will try it this afternoon.

**Reply by Navid200** [reply]
Yes
But, you will need to establish connectivity.
Connectivity is two-way.  The G7 needs to consider xDrip to be the collector and not Dexcom app and this transfer may take a few read cycles.

If you run into trouble, please let me know and I will walk you through the steps required to connect right away.

**Reply by hax90** [reply]
I went ahead and uninstalled Dexcom's app but I haven't had xdrip automatically resolve it's issue yet. What's the process of getting it to pair? Current status is this:
![Screenshot_20240527_203119_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/170917084/4e474c5b-c2ba-4161-a1cc-13998644912a)

I just now did settings > G5/G6/G7/Dex1 Debug Settings > Special Pairing Workaround (checked)

Will that cause it to re-pair? Do I need to forget the bluetooth device and let it repair that way?

**Reply by Navid200** [reply]
Install this version of xDrip: https://github.com/NightscoutFoundation/xDrip/releases/tag/2024.05.15
No need to uninstall first.

Change the last character of the transmitter ID to something else and apply.

Go to the phone Bluetooth settings page and forget all devices starting with DXCM or DX02.

Put all your old used G7 devices inside a metallic box or the microwave or far away.

Raise the volume so that you can hear the pair request ping.  

Now, change the transmitter ID back to the correct value.

Go to the status page and wait for a pair request and approve.

**Reply by Navid200** [reply]
There have been several changes to address more than one problem you have reported here.
Thanks for the report.
Please update to the latest Nightly and tell us if you see any issues:
https://github.com/NightscoutFoundation/xDrip/releases/tag/2024.05.28
